sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,181 @@
+ */
+@Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class ThroughputControlGroup {
+
+    private final static boolean DEFAULT_USE_BY_DEFAULT = false;
+    private final static ThroughputControlMode DEFAULT_CONTROL_MODE = ThroughputControlMode.LOCAL;
+
+    private final String groupName;
+    private final String id;
+    private final CosmosAsyncContainer targetContainer;
+
+    private ThroughputControlMode controlMode;
+    private Integer targetThroughput;
+    private Double targetThroughputThreshold;
+    private boolean useByDefault;
+
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup(String groupName, CosmosAsyncContainer targetContainer, int targetThroughput) {
+        this(groupName, targetContainer, targetThroughput, null);
+    }
+
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup(String groupName, CosmosAsyncContainer targetContainer, double targetThroughputThreshold) {
+        this(groupName, targetContainer, null, targetThroughputThreshold);
+    }
+
+    private ThroughputControlGroup(
+        String groupName,
+        CosmosAsyncContainer targetContainer,
+        Integer targetThroughput,
+        Double targetThroughputThreshold) {
+
+        checkArgument(StringUtils.isNotEmpty(groupName), "Group name can not be null or empty");
+        checkNotNull(targetContainer, "Target container can not be null");
+        checkArgument(targetThroughput == null || targetThroughput > 0, "Target throughput should be greater than 0");
+        checkArgument(
+            targetThroughputThreshold == null || (targetThroughputThreshold > 0 && targetThroughputThreshold <= 1),
+            "Target throughput threshold should between (0, 1]");
+
+        this.groupName = groupName;
+        this.targetContainer = targetContainer;
+        this.targetThroughput = targetThroughput;
+        this.targetThroughputThreshold = targetThroughputThreshold;
+        this.controlMode = DEFAULT_CONTROL_MODE;
+        this.useByDefault = DEFAULT_USE_BY_DEFAULT;
+
+        this.id = this.getId();
+    }
+
+    /**
+     * Get throughput group control mode.
+     *
+     * By default, it will be local control mode.
+     *
+     * @return the {@link ThroughputControlMode}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    ThroughputControlMode getControlMode() {
+        return this.controlMode;
+    }
+
+    /**
+     * Set throughput group control mode.
+     *
+     * By default it will be local control mode.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup setControlMode(ThroughputControlMode controlMode) {
```

Comment:
Are we expecting more than couple values for ThroughputControlMode? If not I will recommend embedding the "mode" into the name of the method. Also for the distributed case you can add a parameter that for that particular case it will take a CosmosAsyncContainer...